aten/src/THNN/generic/SoftMarginCriterion.c

```diff
@@ -7,9 +7,19 @@ void THNN_(SoftMarginCriterion_updateOutput)(
   THTensor *input,
   THTensor *target,
   THTensor *output,
-  bool sizeAverage)
+  bool sizeAverage,
+  bool reduce)
 {
   THNN_CHECK_NELEMENT(input, target);
+
+  if (!reduce) {
+    THTensor_(resizeAs)(output, input);
+
+    TH_TENSOR_APPLY3(real, input, real, target, real, output,
+                     *output_data = log(1. + exp(-*input_data* *target_data));)
```

Comment:
nit: `-*input_data * *target_data` (extra space around the multiplication `*` but it's okay either way)